src/valuation_spectrum.lean

```diff
@@ -1,20 +1,20 @@
 import valuations 
 import analysis.topology.topological_space
 import data.finsupp
-import for_mathlib.quotient_group
+import group_theory.quotient_group
 import valuation_universe 
 
 universes u u1 u2
 
-open valuation 
+open valuation quotient_group
 
 definition Spv (R : Type u) [comm_ring R] := 
 {ineq : R → R → Prop // ∃ (Γ : Type u) [linear_ordered_comm_group Γ],
   by exactI ∃ (v : valuation R Γ), ∀ r s : R, ineq r s ↔ v r ≤ v s}
 
 namespace Spv 
 
-set_option class.instance_max_depth 100
+set_option class.instance_max_depth 41
```

Comment:
Beautiful!

Reply:
I hesitated for a long time between this optimal 41, and the neighboring 42.

Reply:
My impression is that this is the sort of problem which can be fixed with a judicious "instance short-cut" (i.e. explicitly making an instance rather than letting type class inference do it for you). The question is -- which instance?